app/models/task.py

```diff
@@ -1,5 +1,34 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from sqlalchemy import ForeignKey
 from ..db import db
+from datetime import datetime
+from typing import Optional
+
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    description: Mapped[str]
+    completed_at: Mapped[Optional[datetime]]
+    goal_id: Mapped[Optional[int]] = db.Column(db.Integer, db.ForeignKey("goal.id"))
+    goal: Mapped[Optional["Goal"]] = db.relationship("Goal", back_populates="tasks")
```

Comment:
```suggestion
    goal: Mapped[Optional["Goal"]] = relationship("Goal", back_populates="tasks")
```

Opt for using the declarative mapping style, not the classical mapping style